snuba/subscriptions/scheduler.py

```diff
@@ -321,6 +328,36 @@ def __reset_builder(self) -> None:
             # We are transitioning between jittered and immediate mode. We must use the delegate builder.
             self.__builder = self.__delegate_builder
 
+    def __filter_subscriptions(self) -> List[Subscription]:
+
+        # only generic metrics storage sets are currently sliced
+        if (
+            self.__entity_key == EntityKey.GENERIC_METRICS_SETS
+            or self.__entity_key == EntityKey.GENERIC_METRICS_DISTRIBUTIONS
```

Comment:
can we check if the storage set for the given entity key is sliced using the global logic rather than hardcoding these here?

the entity key is constant after initialization so we should only have to test for that once

Reply:
makes sense, changed